src/persist/src/indexed/mod.rs

```diff
@@ -841,20 +840,23 @@ impl<L: Log, B: Blob> Indexed<L, B> {
                     .map(|((k, v), ts, diff)| ((k.to_vec(), v.to_vec()), ts, diff))
                     .collect();
 
-                if listen_fns.len() == 1 {
-                    listen_fns[0].0(ListenEvent::Records(updates));
+                if listeners.len() == 1 {
+                    let events = ListenEvent::Records(updates);
+                    if let Err(crossbeam_channel::SendError(_)) = listeners[0].send(events) {}
```

Comment:
I think if the listener fails we should unregister it here. thoughts?

Reply:
yeah i think that's perfect, i've been thinking the same thing. let's add a test and document it in the StreamReadHandle::listen rustdoc

Reply:
cool cool -- I'll do this in a separate pr to keep the refactor and the functionality changes distinct.

Reply:
leave a TODO then please